lib/auto_api/property.ex

```diff
@@ -93,10 +93,29 @@ defmodule AutoApi.Property do
   """
   @spec to_bin(__MODULE__.t(), spec()) :: binary()
   def to_bin(%__MODULE__{} = prop, spec) do
-    wrap_with_size(prop, :data, &data_to_bin(&1, spec)) <>
-      wrap_with_size(prop, :timestamp, &timestamp_to_bin/1) <>
-      wrap_with_size(prop, :failure, &failure_to_bin/1) <>
-      wrap_with_size(prop, :availability, &availability_to_bin/1)
+    try do
+      wrap_with_size(prop, :data, &data_to_bin(&1, spec)) <>
+        wrap_with_size(prop, :timestamp, &timestamp_to_bin/1) <>
+        wrap_with_size(prop, :failure, &failure_to_bin/1) <>
+        wrap_with_size(prop, :availability, &availability_to_bin/1)
+    rescue
+      error ->
+        Logger.error([
+          "Not able to serialize value for spec #{inspect(spec)} to binary ",
+          inspect(error),
+          " stacktrace: #{inspect Process.info(self(), :current_stacktrace)}"
+        ])
+
+        failure = %__MODULE__{
```

Comment:
So we do put a failure message? Hm.
Not 100% sure about it, but maybe we can change it once we will have a proper error/data validation in place?

Good for me right now 👍🏻

Reply:
Yeah I was also not happy with changing the property to failure seemingly out of no where.

However If we keep the XV in mind this is a good solution since we can handle all errors gracefully and still be able to give exact detail what was wrong. right now it's hit and miss during the integration phase.